File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 08/Go/main.go
package main

import "fmt"

type Empregado struct {
    Nome    string
    Cargo   string
    Salario float64
}

type Empresa struct {
    Empregados []Empregado
}

func (e *Empresa) AdicionarEmpregado(empregado Empregado) {
    e.Empregados = append(e.Empregados, empregado)
}

func main() {
    empresa := Empresa{}
    empresa.AdicionarEmpregado(Empregado{"João", "Desenvolvedor", 3000.00})
    empresa.AdicionarEmpregado(Empregado{"Maria", "Gerente", 5000.00})

    for _, e := range empresa.Empregados {
        fmt.Printf("%s - %s - %.2f\n", e.Nome, e.Cargo, e.Salario)
    }
}

File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 08/Python/main.py
class Empregado:
    def __init__(self, nome, cargo, salario):
        self.nome = nome
        self.cargo = cargo
        self.salario = salario

class Empresa:
    def __init__(self):
        self.empregados = []

    def adicionar_empregado(self, empregado):
        self.empregados.append(empregado)

empresa = Empresa()
empresa.adicionar_empregado(Empregado("João", "Desenvolvedor", 3000.00))
empresa.adicionar_empregado(Empregado("Maria", "Gerente", 5000.00))

for empregado in empresa.empregados:
    print(f"{empregado.nome} - {empregado.cargo} - {empregado.salario}")
